LanterFish fix for categories query

klas opened the pull request:
This just changes query from string to Joomla query builder so that LanterFish can do its job. Previous attempt was missing $db->setQuery($query); and was since executing some other sql. Sorry for that :)

xtech86 commented:
Hello,

What's your logic for removing:

$query = "SELECT c.id"

and adding in a double:

$catlist = $db->loadColumn();
$catlist = $db->loadColumn();

Thanks
Tony

klas commented:
$query = "SELECT c.id" is not removed, this is github showing it in weird way, the whole query as it was is commented out and replaced by Joomla query builder version, see full file https://github.com/klas/JEvents/blob/master/component/site/libraries/dbmodel.php#L113

Good catch for double $catlist = $db->loadColumn(); , that was a typo, fixed.
